apps/api/src/app/preferences/preferences.spec.ts

```diff
@@ -0,0 +1,816 @@
+      expect(subscriberWorkflowPreferences._subscriberId).to.equal(subscriberId);
+      expect(subscriberWorkflowPreferences.actor).to.equal(PreferencesActorEnum.SUBSCRIBER);
+    });
+
+    it('should update preferences', async function () {
+      let workflowPreferences = await upsertPreferences.execute(
+        UpsertPreferencesCommand.create({
+          preferences: {
+            workflow: {
+              defaultValue: false,
+              readOnly: false,
+            },
+            channels: {
+              in_app: {
+                defaultValue: false,
+                readOnly: false,
+              },
+              sms: {
+                defaultValue: false,
+                readOnly: false,
+              },
+              email: {
+                defaultValue: false,
+                readOnly: false,
+              },
+              push: {
+                defaultValue: false,
+                readOnly: false,
+              },
+              chat: {
+                defaultValue: false,
+                readOnly: false,
+              },
+            },
+          },
+          actor: PreferencesActorEnum.WORKFLOW,
+          environmentId: session.environment._id,
+          organizationId: session.organization._id,
+          templateId: workflowId,
+        })
+      );
+
+      expect(workflowPreferences._environmentId).to.equal(session.environment._id);
+      expect(workflowPreferences._organizationId).to.equal(session.organization._id);
+      expect(workflowPreferences._templateId).to.equal(workflowId);
+      expect(workflowPreferences._userId).to.be.undefined;
+      expect(workflowPreferences._subscriberId).to.be.undefined;
+      expect(workflowPreferences.actor).to.equal(PreferencesActorEnum.WORKFLOW);
+
+      workflowPreferences = await upsertPreferences.execute(
+        UpsertPreferencesCommand.create({
+          preferences: {
+            workflow: {
+              defaultValue: false,
+              readOnly: true,
+            },
+            channels: {
+              in_app: {
+                defaultValue: false,
+                readOnly: false,
+              },
+              sms: {
+                defaultValue: false,
+                readOnly: false,
+              },
+              email: {
+                defaultValue: false,
+                readOnly: false,
+              },
+              push: {
+                defaultValue: false,
+                readOnly: false,
+              },
+              chat: {
+                defaultValue: false,
+                readOnly: false,
+              },
+            },
+          },
+          actor: PreferencesActorEnum.WORKFLOW,
+          environmentId: session.environment._id,
+          organizationId: session.organization._id,
+          templateId: workflowId,
+        })
+      );
+
+      expect(workflowPreferences.preferences.workflow.readOnly).to.be.true;
+    });
+
+    it('should validate preferences', async function () {
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
🤓 **nitpick (non-blocking):** Please consider separating into distinct tests; it is easier to tell what's failing when separate tests cover one piece of functionality‏